netket/abstract_variational_driver.py

```diff
@@ -225,20 +237,28 @@ def run(
         callbacks = _to_iterable(callback)
         callback_stop = False
 
-        with tqdm(
-            self.iter(n_iter, step_size), total=n_iter, disable=not show_progress
-        ) as itr:
-            for step in itr:
-
+        with tqdm(total=n_iter, disable=not show_progress) as pbar:
+            old_step_value = self.step_value
+            for step in self.iter(n_iter, step_size):
                 log_data = self.estimate(obs)
 
                 # if the cost-function is defined then report it in the progress bar
+                # and to the loggers
                 if self._loss_stats is not None:
-                    itr.set_postfix_str(self._loss_name + "=" + str(self._loss_stats))
+                    pbar.set_postfix_str(self._loss_name + "=" + str(self._loss_stats))
                     log_data[self._loss_name] = self._loss_stats
 
-                for logger in loggers:
-                    logger(self.step_count, log_data, self.machine)
+                if len(loggers) > 0:
+                    # this function can be overriden by drivers to append anything
+                    # they want to the logged data
+                    self._log_additional_data(log_data, step)
+
+                    for logger in loggers:
+                        logger(self.step_count, log_data, self.machine)
+
+                # Update the progress bar
+                pbar.update(self.step_value - old_step_value)
+                old_step_value = self.step_value
```

Comment:
Maybe we should have some rounding for display purposes? Otherwise, we end up with floating point times such as
```
 99%|███▉| 1.9700000000000015/2.0
```
and even
```
100%|████| 2.0000000000000013/2.0
```
Not much of a problem, but it doesn't look very nice.